### PR TITLE
[None][feat] enable TRTLLM-Gen internal routing

### PR DESCRIPTION
**Pull request by tcherckez-nvidia**
Route matched noaux NVFP4 MoE patterns to TRTLLM-Gen internal routing by default when possible.

Keep the external-routing path for all-to-all cases, preserve routing_bias dtype for TRTLLM-Gen, and cover direct and EP-masked matcher shapes.

## Summary by CodeRabbit

* **New Features**
  * Added internal routing optimization for MoE (Mixture of Experts) operations
  * Enabled multi-stream MoE execution for improved performance with independent shared and routed expert branches

* **Tests**
  * Added test coverage for MoE internal routing validation

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/TensorRT-LLM/pull/13997)

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by tcherckez-nvidia** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

File: tensorrt_llm/_torch/auto_deploy/transform/library/fused_moe.py [hunk 1]
```diff
@@ -2697,6 +2873,13 @@ class FuseNVFP4MoeConfig(TransformConfig):
             "before TRTLLM-Gen shuffle+interleave. Only used when backend='trtllm_gen'."
         ),
     )
+    enable_trtllm_gen_internal_routing: bool = Field(
+        default=True,
+        description=(
+            "If True and backend='trtllm_gen', pass DeepSeekV3 router logits and routing bias "
```
Comment (galagam) [inline tensorrt_llm/_torch/auto_deploy/transform/library/fused_moe.py]: Why is `DeepSeekV3` mentioned here specifically?

**Review by galagam — APPROVED** [review]
LGTM

File: tensorrt_llm/_torch/auto_deploy/transform/library/fused_moe.py (hunk 1 above)
Comment (tcherckez-nvidia) [reply tensorrt_llm/_torch/auto_deploy/transform/library/fused_moe.py]: type of MoE

**Comment by tcherckez-nvidia** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

File: tensorrt_llm/_torch/auto_deploy/transform/library/fused_moe.py (hunk 1 above)
Comment (tcherckez-nvidia) [inline tensorrt_llm/_torch/auto_deploy/transform/library/fused_moe.py]:
```suggestion
            "If True and backend='trtllm_gen', pass router logits and routing bias "
```

**Comment by tcherckez-nvidia** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by tcherckez-nvidia** [comment]
/bot run --stage-list "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"
